src/materials/ADComputeISoilStress.C

```diff
@@ -0,0 +1,642 @@
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_2",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_3",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_4",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_5",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("taumax",
+                                       "The ultimate shear strength of "
+                                       "the soil layers. Required for "
+                                       "GQ/H model");
+    // params required for thin_layer contact model soil_type = "thin_layer"
+    params.addParam<std::vector<Real>>("friction_coefficient",
+                                       "Friction coefficients of the thin layers.");
+    params.addParam<std::vector<Real>>("hardening_ratio",
+                                       "Post-yield hardening ratios of the layers."););
+
+template <ComputeStage compute_stage>
+ADComputeISoilStress<compute_stage>::ADComputeISoilStress(const InputParameters & parameters)
+  : ADComputeFiniteStrainElasticStress<compute_stage>(parameters),
+    _base_models(),
+    _stress_model(),
+    _stress_model_old(),
+    _yield_stress(), // *** YIELD STRAIN NOT YIELD STRESS ***
+    _youngs(),
+    _soil_layer_variable(coupledValue("layer_variable")),
+    _layer_ids(getParam<std::vector<unsigned int>>("layer_ids")),
+    _wave_speed_calculation(getParam<bool>("wave_speed_calculation")),
+    _poissons_ratio(getParam<std::vector<Real>>("poissons_ratio")),
+    _density(_wave_speed_calculation ? &getMaterialProperty<Real>("density") : NULL),
+    _b_exp(getParam<Real>("b_exp")),
+    _p_ref(getParam<std::vector<Real>>("p_ref")),
+    _a0(getParam<Real>("a0")),
+    _a1(getParam<Real>("a1")),
+    _a2(getParam<Real>("a2")),
+    _p0(getParam<Real>("tension_pressure_cut_off")),
+    _pressure_dependency(getParam<bool>("pressure_dependency")),
+    _strength_pressure_correction(1.0),
+    _stiffness_pressure_correction(1.0),
+    _shear_wave_speed(_wave_speed_calculation ? &declareProperty<Real>("shear_wave_speed") : NULL),
```

Comment:
Same question about independence from nonlinear dofs. This question applies to any place I see declareProperty vs declareADProperty and getMaterialProperty vs getADMaterialProperty

Reply:
Yes, the shear wave speed and P wave speed are material properties that are computed based on the user provided data. It does not vary with nonlinear dofs.